### PR TITLE
Added solution for HW_5.2

### PR DESCRIPTION
**Pull request by lord-dwarf**

### CODE REVIEW

File: src/main/java/core/basesyntax/Plane.java [hunk 1]
```diff
@@ -7,5 +7,64 @@
...
+            this.maxCargo = maxCargo;
+            return this;
+        }
+
+        public Plane build() {
+            return new Plane(this);
+        }
+    }
+
+    public static void main(String[] args) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Plane.java]: lets remove this from the Plane class. this method was needed for self test

Comment (lord-dwarf) [reply src/main/java/core/basesyntax/Plane.java]: done

File: src/main/java/core/basesyntax/Plane.java [hunk 2]
```diff
@@ -7,5 +7,57 @@
...
+
+    public static class PlaneBuilder {
+        private String name;
+        private String type;
+        private int maxSpeed;
+        private double maxCargo;
+        private int maxSeats;
+
+        private PlaneBuilder() {
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Plane.java]: try to create PlaneBuilder, set few fields and then create Plane out of it 
pass your code as answer for this comment( don't push it to git)

Comment (lord-dwarf) [reply src/main/java/core/basesyntax/Plane.java]: PlaneBuilder pb = new PlaneBuilder();
        pb.setMaxCargo(10);
        pb.setMaxSeats(23);
        Plane newPlane = pb.build();

Comment (lord-dwarf) [reply src/main/java/core/basesyntax/Plane.java]: originally I had public static factory method to create builder, I eliminate it to conform to other solutions and forgot to delete it - yes, I got that it prevents from creating PlaneBuilder outside of this class, but tests didn't complain. I reverted it

Comment (lord-dwarf) [reply src/main/java/core/basesyntax/Plane.java]: added public method builder() to access private constructor of PlaneBuilder and pushed it to git

File: src/main/java/core/basesyntax/Plane.java [hunk 3]
```diff
@@ -7,5 +7,61 @@
...
+
+    public static class PlaneBuilder {
+        private String name;
+        private String type;
+        private int maxSpeed;
+        private double maxCargo;
+        private int maxSeats;
+
+        public PlaneBuilder() {
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Plane.java]: so, do we have to create public constructor with no parameters?)

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Plane.java]: or we can create it like this Plane.PlaneBuilder?

Comment (lord-dwarf) [reply src/main/java/core/basesyntax/Plane.java]: No, we don't need it. Sorry, I tried to do it another way (elegant builder) and got confused. Solved.

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**